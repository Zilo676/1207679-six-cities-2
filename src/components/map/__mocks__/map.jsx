import React from "react";

const Map = () => {
  return <></>;
};

export {Map};
export default Map;
