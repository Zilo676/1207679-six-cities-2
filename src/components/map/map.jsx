import React from "react";
import PropTypes from 'prop-types';
import leaflet from 'leaflet';

const MapConfiguration = {
  ZOOM: 12,
  ICON_URL: `img/pin.svg`,
  ICON_SIZE: [30, 30],
  TILE_LAYER: `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
  TILE_ATTRIBUTE: `&copy; <a href="htps://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`,
  CITY: [52.38333, 4.9],
};

class Map extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this._initializeMap();
  }

  _initializeMap() {
    const map = leaflet.map(`map`, {
      center: MapConfiguration.CITY,
      zoom: MapConfiguration.ZOOM,
      zoomControl: false,
      marker: true
    });

    map.setView(MapConfiguration.CITY, MapConfiguration.ZOOM);
    leaflet
      .tileLayer(MapConfiguration.TILE_LAYER, {
        attribution: MapConfiguration.TILE_ATTRIBUTE
      })
      .addTo(map);

    const icon = leaflet.icon({
      iconUrl: MapConfiguration.ICON_URL,
      iconSize: MapConfiguration.ICON_SIZE
    });

    const offersСoordinates = this.props.coordinates;
    offersСoordinates.forEach((it) => {
      leaflet
        .marker(it, {icon})
        .addTo(map);
    });
  }

  render() {
    return <div className="cities__map map" id="map" />;
  }
}

// TODO: Написать кастомный пропс с длинной массива в 2 элемента
Map.propTypes = {
  coordinates: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
};

export {Map};
