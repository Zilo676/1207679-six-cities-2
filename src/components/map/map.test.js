import React from 'react';
import renderer from 'react-test-renderer';

import {Map} from './map.jsx';

const coordinates = [
  {
    latitude: 52.35514938496378,
    longitude: 4.673877537499948,
    zoom: 8
  },
  {
    latitude: 52.35514938496378,
    longitude: 4.673877537499948,
    zoom: 8
  },
];

const city = {
  name: `Amsterdam`,
  location: {
    latitude: 52.370216,
    longitude: 4.895168,
    zoom: 10
  }
};

// https://stackoverflow.com/questions/40852131/refs-are-null-in-jest-snapshot-tests-with-react-test-renderer
const createNodeMock = () => document.createElement(`div`);

it(`Correctly rendered`, () => {
  const tree = renderer.create(<Map
    location={coordinates}
    city={city}
    cssClass={`abc`}
  />, {createNodeMock}).toJSON();
  expect(tree).toMatchSnapshot();
});


