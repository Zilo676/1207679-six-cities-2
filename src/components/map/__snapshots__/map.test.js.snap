// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered 1`] = `
<div
  className="abc__map map"
  id="map"
/>
`;
